### PR TITLE
[EC-508] SCIM CQRS Refactor - Users/GetList

### PR DESCRIPTION
**Pull request by r-tome**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [X] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective
Refactor the SCIM API to use the CQRS pattern which implies that each action must be isolated in a specific command to do only one specific task.
Added unit tests to check that each command completes its task.

## Code changes
* **bitwarden-server.sln:** Added new unit tests project Scim.Test
* **bitwarden_license/src/Scim/Commands/Users/Interfaces/*:** SCIM user endpoints command interfaces
* **bitwarden_license/src/Scim/Commands/Users/*:** SCIM user endpoints command implementations
* **bitwarden_license/src/Scim/Controllers/v2/UsersController.cs:** Updated to use commands and handle exceptions
* **bitwarden_license/src/Scim/Startup.cs:** Registered commands for injection
* **bitwarden_license/test/Scim.Test/Commands/Users/*:** SCIM user command unit tests
* **bitwarden_license/test/Scim.Test/Scim.Test.csproj:** New unit test project for SCIM
* **bitwarden_license/test/Scim.Test/packages.lock.json:** Scim.Test project package dependencies
* **src/Core/Exceptions/BadRequestException.cs:** Added parameterless constructor
* **src/Core/Exceptions/ConflictException.cs:** Added a new Exception type for Conflict
* **src/Core/Exceptions/NotFoundException.cs:** Added parameterless constructor

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

**Review by eliykat — CHANGES_REQUESTED** [review]
I merged `master` into this branch and resolved conflicts to clean up the diff. But I assume all the other PRs have the same issue. Could you please resolve conflicts across all PRs, then I should be able to sit down & tear through them, now that we have the basic patterns down.

File: bitwarden_license/src/Scim/Controllers/v2/UsersController.cs [hunk 1]
```diff
@@ -34,6 +35,7 @@ public class UsersController : Controller
         IOrganizationService organizationService,
         IScimContext scimContext,
         IOptions<ScimSettings> scimSettings,
```
[the remark this answers is not in the document]
Comment (r-tome) [reply bitwarden_license/src/Scim/Controllers/v2/UsersController.cs]: I've updated all branches and removed unused dependencies.

**Review by eliykat — APPROVED** [review]

File: bitwarden_license/src/Scim/Users/GetUserQuery.cs [hunk 2]
```diff
@@ -1,6 +1,6 @@
 ﻿using Bit.Core.Exceptions;
```
Comment (eliykat) [inline bitwarden_license/src/Scim/Users/GetUserQuery.cs:1]: Let's just pull this `GetUserQuery` back out into the controller (i.e. leave it how it was). It was an early one that I approved before we had that discussion with Oscar about when it's worth using a query. In this case, the repository call is basically the query, there's so little code here that we can just have it in the controller. (sorry for the busywork)

Comment (eliykat) [reply bitwarden_license/src/Scim/Users/GetUserQuery.cs:1]: You can do this in a separate PR though, if you want.

**Review by eliykat — APPROVED** [review]

File: bitwarden_license/src/Scim/Users/GetUserQuery.cs (hunk 2 above)
Comment (r-tome) [reply bitwarden_license/src/Scim/Users/GetUserQuery.cs:1]: I'll do that on the next PR when merging to master. Thanks!

**Pull request merged**